climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,728 @@
+
+
+    def plot(self):
+        """
+        Plot the distributions of the parameters of the uncertainty variable.
+
+        Returns
+        -------
+        fig, ax: matplotlib.pyplot.figure, matplotlib.pyplot.axes
+            The figure and axis handle of the plot.
+
+        """
+        # zip_longest(flat_axes, cols, fillvalue=None):
+
+        nplots = len(self.distr_dict)
+        nrows, ncols = int(nplots / 3) + 1, min(nplots, 3)
+        fig, axis = plt.subplots(nrows=nrows, ncols=ncols, figsize=(20, 16))
+        if nplots > 1:
+            flat_axes = axis.flatten()
+        else:
+            flat_axes = [axis]
+        for ax, name_distr in zip_longest(flat_axes,
+                                    self.distr_dict.items(), 
+                                    fillvalue=None):
+            if name_distr is None:
+                ax.remove()
+                continue
+            (param_name, distr) = name_distr
+            x = np.linspace(distr.ppf(0.001), distr.ppf(0.999), 100)
+            ax.plot(x, distr.pdf(x), label=param_name)
+            ax.legend()
+        return fig, axis
+
+
+    def evaluate(self, uncvar_kwargs):
+        """
+        Evaluate the uncertainty variable.
+
+        Parameters
+        ----------
+        uncvar_kwargs : dict
+            These parameters will be passed to self.unc_var.
+            They must be the input parameters of the uncertainty variable .
+
+        Returns
+        -------
+
+            Evaluated uncertainty variable
+
+        """
+        return self.unc_var(**uncvar_kwargs)
+    
+    
+    @staticmethod
+    def var_to_uncvar(var):
+        """
+        Returns uncertainty variable with no distribution if var is not
+        an UncVar. Else, returns var.
+
+        Parameters
+        ----------
+        var : Object or else
+
+        Returns
+        -------
+        UncVar
+            var if var is UncVar, else UncVar with var and no distribution.
+
+        """
+
+        if isinstance(var, UncVar):
+            return var
+
+        return UncVar(unc_var=lambda: var, distr_dict={})
+
+
+
+class Uncertainty():
+    """
+    Uncertainty analysis class
+
+    This is the base class to perform uncertainty analysis on the outputs of a
+    climada.engine.impact.Impact() or climada.engine.costbenefit.CostBenefit()
+    object.
+
+    """
+
+    def __init__(self, unc_vars, sample=None, metrics=None,
+                 sensitivity=None):
+        """
+        Initialize Uncertainty
+
+        Parameters
+        ----------
+        unc_vars : dict
+            keys are names and values are climade.engine.uncertainty.UncVar 
+        sample : pd.DataFrame, optional
+            DataFrame of sampled parameter values. Column names must be
+            parameter names (all labels) from all unc_vars.
+            The default is pd.DataFrame().
+        metrics : dict(), optional
+            Dictionnary of the CLIMADA metrics for which the sensitivity
```

Comment:
I  think this would be less precise, as not all attributes of `Impacts` or `CostBenefit` are fit for a sensitivity analysis, but only the outputs of the `.calc` methods. I called these outputs `metrics` (I heard this wording in a few talks on risk modelling and found it rather clear).

Reply:
Is it clearer now?

a0b71670c65e9384119256500fb64a267591cbad